### PR TITLE
Use an all pyproject.toml setup

### PR DESCRIPTION
**Pull request by ocefpaf**
I also moved the linting to ruff.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,75 @@
...
+    {name = "Filipe Fernandes", email = "ocefpaf+erddapy@gmail.com"},
+]
+requires-python = ">=3.8"
+dynamic = [
+  "version",
+]
+dependencies = [
+  "httpx",
+  "pandas>=1",
+]
```
Comment (abkfenris) [inline pyproject.toml:24]: I think a newer version of setuptools can also dynamically pull dependencies from `requirements.txt` to get rid of some duplication.

File: MANIFEST.in
Comment (abkfenris) [inline MANIFEST.in:1]: Is the manifest still necessary with `pyproject.toml`?

File: pyproject.toml (hunk 1 above)
Comment (ocefpaf) [reply pyproject.toml:24]: That would be nice. Do you have an example of it?

File: MANIFEST.in
Comment (ocefpaf) [reply MANIFEST.in:1]: Sadly we do. We have more control over the sdist with the MANIFEST. We can remove it and the sdist will be fine but we would be packaging some junk and making it a bit big. B/c PyPI has some size limitations I like to keep them small.

File: pyproject.toml (hunk 1 above)
Comment (abkfenris) [reply pyproject.toml:24]: https://github.com/gulfofmaine/xpublish-edr/blob/b0c3e877475f77215769dd72afcdc3670b45b22a/pyproject.toml#L27-L33

**Comment by ocefpaf** [comment]
I'll address the docs failures in another PR, this one is already too big for its own sake.. This should be ready to go but I'd like to highlight this commits here in case it causes us problem in the future: https://github.com/ioos/erddapy/pull/297/commits/ccfa937e722ef7e197d9041bda031a96f4e226e9

Hopefully that addresses the memory leak caused by creating too many instances of the class without hurting the cache.

**Comment by ocefpaf** [comment]
@abkfenris thanks for the dynamic requirement file tip. Looks much cleaner now!

**Pull request merged**